source/portable/NetworkInterface/STM32/NetworkInterface.c

```diff
@@ -0,0 +1,2373 @@
+        #endif
+        const uint32_t ulPriority = NVIC_GetPriority( ETH_IRQn ) << ( 8U - ulPrioBits );
+
+        if( ulPriority < configMAX_SYSCALL_INTERRUPT_PRIORITY )
+        {
+            FreeRTOS_debug_printf( ( "prvEthConfigInit: Incorrectly set ETH_IRQn priority\n" ) );
+            NVIC_SetPriority( ETH_IRQn, configMAX_SYSCALL_INTERRUPT_PRIORITY >> ( 8U - ulPrioBits ) );
+        }
+
+        if( NVIC_GetEnableIRQ( ETH_IRQn ) == 0 )
+        {
+            FreeRTOS_debug_printf( ( "prvEthConfigInit: ETH_IRQn was not enabled by application\n" ) );
+            HAL_NVIC_EnableIRQ( ETH_IRQn );
+        }
+
+        #ifdef niEMAC_STM32FX
+            configASSERT( __HAL_RCC_ETH_IS_CLK_ENABLED() != 0 );
+        #elif defined( STM32H5 )
+            configASSERT( __HAL_RCC_ETH_IS_CLK_ENABLED() != 0 );
+            configASSERT( __HAL_RCC_ETHTX_IS_CLK_ENABLED() != 0 );
+            configASSERT( __HAL_RCC_ETHRX_IS_CLK_ENABLED() != 0 );
+        #elif defined( STM32H7 )
+            configASSERT( __HAL_RCC_ETH1MAC_IS_CLK_ENABLED() != 0 );
+            configASSERT( __HAL_RCC_ETH1TX_IS_CLK_ENABLED() != 0 );
+            configASSERT( __HAL_RCC_ETH1RX_IS_CLK_ENABLED() != 0 );
+        #endif
+    }
+
+    return xResult;
+}
+
+/*---------------------------------------------------------------------------*/
+
+static void prvInitMacAddresses( ETH_HandleTypeDef * pxEthHandle,
+                                 NetworkInterface_t * pxInterface )
+{
+    ETH_MACFilterConfigTypeDef xFilterConfig;
+
+    ( void ) HAL_ETH_GetMACFilterConfig( pxEthHandle, &xFilterConfig );
+    xFilterConfig.ReceiveAllMode = DISABLE;
+    xFilterConfig.HachOrPerfectFilter = ENABLE;
+    xFilterConfig.SrcAddrFiltering = DISABLE;
+    xFilterConfig.SrcAddrInverseFiltering = DISABLE;
+    xFilterConfig.ControlPacketsFilter = ETH_CTRLPACKETS_BLOCK_ALL;
+    xFilterConfig.BroadcastFilter = ENABLE;
+    xFilterConfig.PassAllMulticast = DISABLE;
+    xFilterConfig.DestAddrInverseFiltering = DISABLE;
+    xFilterConfig.HashMulticast = ENABLE;
+    xFilterConfig.HashUnicast = ENABLE;
+    xFilterConfig.PromiscuousMode = DISABLE;
+    ( void ) HAL_ETH_SetMACFilterConfig( pxEthHandle, &xFilterConfig );
+
+    NetworkEndPoint_t * pxEndPoint;
+
+    for( pxEndPoint = FreeRTOS_FirstEndPoint( pxInterface ); pxEndPoint != NULL; pxEndPoint = FreeRTOS_NextEndPoint( pxInterface, pxEndPoint ) )
+    {
+        prvAddAllowedMACAddress( pxInterface, pxEndPoint->xMACAddress.ucBytes );
+    }
+
+    #if ipconfigIS_ENABLED( ipconfigUSE_IPv4 )
+        #if ipconfigIS_ENABLED( ipconfigUSE_MDNS )
+            prvAddAllowedMACAddress( pxInterface, xMDNS_MacAddress.ucBytes );
+        #endif
+        #if ipconfigIS_ENABLED( ipconfigUSE_LLMNR )
+            prvAddAllowedMACAddress( pxInterface, xLLMNR_MacAddress.ucBytes );
+        #endif
+    #endif
+
+    #if ipconfigIS_ENABLED( ipconfigUSE_IPv6 )
+        prvAddAllowedMACAddress( pxInterface, pcLOCAL_ALL_NODES_MULTICAST_MAC.ucBytes );
```

Comment:
This gives me: `error: request for member 'ucBytes' in something not a structure or union`

I assume it was supposed to look like this:
```suggestion
        prvAddAllowedMACAddress( pxInterface, pcLOCAL_ALL_NODES_MULTICAST_MAC );
```